src/com/twitter/intellij/pants/service/project/PantsSystemProjectResolver.java

```diff
@@ -147,7 +149,16 @@ private void checkForDifferentPantsExecutables(@NotNull ExternalSystemTaskId id,
     final ProjectData projectData = new ProjectData(
       PantsConstants.SYSTEM_ID,
       executor.getProjectName(),
-      executor.getBuildRoot().getPath() + "/.idea/pants-projects/" + executor.getProjectRelativePath(),
+      Paths.get(
+        executor.getBuildRoot().getPath(),
+        ".idea",
+        "pants-projects",
+        // Use a timestamp hash to avoid module creation dead lock
+        // when overlapping targets were imported into multiple projects
+        // from the same Pants repo.
+        DigestUtils.sha1Hex(Long.toString(System.currentTimeMillis())),
```

Comment:
Is there something stable you could use (fingerprint of the target roots?) or would those all still cause collisions?

Reply:
Right, having the same path means the root cause still remains. That said, the SHA is only used to create the project, and module files will stay the same at project refresh.